### ISSUE TITLE
Automated workers removing road outside own territory

### ISSUE THREAD

**Issue by calloatti** [LoneGazebo/Community-Patch-DLL#10703]
_1. Mod Version (X.Y.Z). Current Version: 4.5.0_

4.5.1

_2. DirectX Version_

_3. List of Other Mods_

_4. Describe the Issue_

Automated workers remove roads that are outside of owned land. This should not happen IMHO. Also maybe tone down a bit the automated road removal or add an option to disable it.

______________________________________________________________________________________
_5. Save Game From 1 Turn Before (ALWAYS ATTACH THIS IF POSSIBLE)_

_6. Logs (ALWAYS ATTACH THESE IF POSSIBLE)_

_7. CvMiniDump.dmp File (ATTACH IF REPORTING A GAME CRASH)_

_8. Steps to reproduce the Issue (Optional)_

_9. Screenshots of the Issue (Optional)_

**Comment by usadefcon1** [comment]
I agree with this, I built that road for a reason. As a temporary fix you can enable "Automated Worker's Don't Replace Improvements" in the Game Options menu.

**Comment by calloatti** [comment]
This automated worker is removing a road over there... A bit overzealous I would say.

![image](https://github.com/LoneGazebo/Community-Patch-DLL/assets/12817862/89acef02-1add-4454-9029-df108ee7aa9a)

**Comment by axatin** [comment]
@KungCheops

**Comment by KungCheops** [comment]
> This automated worker is removing a road over there... A bit overzealous I would say.
> 
> ![image](https://private-user-images.githubusercontent.com/12817862/310023445-89acef02-1add-4454-9029-df108ee7aa9a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5ODk3MzQsIm5iZiI6MTcwOTk4OTQzNCwicGF0aCI6Ii8xMjgxNzg2Mi8zMTAwMjM0NDUtODlhY2VmMDItMWFkZC00NDU0LTkwMjktZGYxMDhlZTdhYTlhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA5VDEzMDM1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzZjQ2MGNhMThmMWQwYzg3MDRiMGY0YTM4N2U0YjI4ZjMyMGM5MTE0MWYyOGQ2ZTgyNzZjNDg1MzVkZDdiMTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gZhSDXuwqaUOa5BPvRrnPlg7HrBH9Fl-y4nEuYxJ-mo)

How did you end up paying maintenance for that road?

> _1. Mod Version (X.Y.Z). Current Version: 4.5.0_
> 
> 4.5.1
> 
> _2. DirectX Version_
> 
> _3. List of Other Mods_
> 
> _4. Describe the Issue_
> 
> Automated workers remove roads that are outside of owned land. This should not happen IMHO. Also maybe tone down a bit the automated road removal or add an option to disable it.
> 
> _5. Save Game From 1 Turn Before (ALWAYS ATTACH THIS IF POSSIBLE)_
> 
> _6. Logs (ALWAYS ATTACH THESE IF POSSIBLE)_
> 
> _7. CvMiniDump.dmp File (ATTACH IF REPORTING A GAME CRASH)_
> 
> _8. Steps to reproduce the Issue (Optional)_
> 
> _9. Screenshots of the Issue (Optional)_

Why should this not happen?

**Comment by KungCheops** [comment]
Hmm, guess I can tweak it a bit

**Comment by calloatti** [comment]
> > This automated worker is removing a road over there... A bit overzealous I would say.
> > ![image](https://private-user-images.githubusercontent.com/12817862/310023445-89acef02-1add-4454-9029-df108ee7aa9a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5ODk3MzQsIm5iZiI6MTcwOTk4OTQzNCwicGF0aCI6Ii8xMjgxNzg2Mi8zMTAwMjM0NDUtODlhY2VmMDItMWFkZC00NDU0LTkwMjktZGYxMDhlZTdhYTlhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA5VDEzMDM1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzZjQ2MGNhMThmMWQwYzg3MDRiMGY0YTM4N2U0YjI4ZjMyMGM5MTE0MWYyOGQ2ZTgyNzZjNDg1MzVkZDdiMTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gZhSDXuwqaUOa5BPvRrnPlg7HrBH9Fl-y4nEuYxJ-mo)
> 
> How did you end up paying maintenance for that road?

https://forums.civfanatics.com/resources/caravans-build-roads.25370/

> 
> > Automated workers remove roads that are outside of owned land. This should not happen IMHO. Also maybe tone down a bit the automated road removal or add an option to disable it.
> 
> Why should this not happen?

You are right, the problem seems to be the interaction with the caravans build roads mod

**Comment by KungCheops** [comment]
> > > This automated worker is removing a road over there... A bit overzealous I would say.
> > > ![image](https://private-user-images.githubusercontent.com/12817862/310023445-89acef02-1add-4454-9029-df108ee7aa9a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDk5ODk3MzQsIm5iZiI6MTcwOTk4OTQzNCwicGF0aCI6Ii8xMjgxNzg2Mi8zMTAwMjM0NDUtODlhY2VmMDItMWFkZC00NDU0LTkwMjktZGYxMDhlZTdhYTlhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMDklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzA5VDEzMDM1NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzZjQ2MGNhMThmMWQwYzg3MDRiMGY0YTM4N2U0YjI4ZjMyMGM5MTE0MWYyOGQ2ZTgyNzZjNDg1MzVkZDdiMTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gZhSDXuwqaUOa5BPvRrnPlg7HrBH9Fl-y4nEuYxJ-mo)
> > 
> > 
> > How did you end up paying maintenance for that road?
> 
> https://forums.civfanatics.com/resources/caravans-build-roads.25370/
> 
> > > Automated workers remove roads that are outside of owned land. This should not happen IMHO. Also maybe tone down a bit the automated road removal or add an option to disable it.
> > 
> > 
> > Why should this not happen?
> 
> You are right, the problem seems to be the interaction with the caravans build roads mod

I decided to make road removal far from friendly territory considered unsafe. Seems reasonable in any case.
